core/src/dird/vbackup.cc

```diff
@@ -208,25 +213,41 @@ bool DoNativeVbackup(JobControlRecord* jcr)
       return false;
     }
 
-    jobids = strdup(jobids_ctx.GetAsString().c_str());
+    jobids = jobids_ctx.GetAsString();
   }
 
-  Jmsg(jcr, M_INFO, 0, _("Consolidating JobIds %s\n"), jobids);
+  {
+    // check if at least one of the jobids has files in it
+    RestoreContext rx;
+    if (PrepareBootstrapFile(jcr, jobids, rx) == 0) {
+      Jmsg(jcr, M_WARNING, 0,
+           _("JobIds %s contain %d files, cannot continue\n"), jobids.data(),
+           NrFiles);
+      return false;
+    } else {
+      Jmsg(jcr, M_INFO, 0, _("Consolidating JobIds %s containing %d files\n"),
+           jobids.data(), NrFiles);
+    }
+    if (!CreateBootstrapFile(jcr, jobids, rx)) {
+      Jmsg(jcr, M_FATAL, 0, _("Could not create bootstrap file\n"));
+      return false;
+    }
+  }
 
   // Find oldest Jobid, get the db record and find its level
-  p = strchr(jobids, ','); /* find oldest jobid */
+  p = strchr(const_cast<char*>(jobids.data()), ','); /* find oldest jobid */
```

Comment:
at second thought: all of this pointer trickery is used, because we get a comma-separated list of jobids in a string and we need the first and the last of these.
I propose to have a function `vector<std::string> split_string(const std::string& str, const std:string& delim)` that splits `str` at `delim` and puts the pieces into a vector it returns.
We can then simply do:
```
auto jobid_list = split_string(jobids, ',');
jcr->impl->previous_jr = JobDbRecord{};
jcr->impl->previous_jr.JobId = str_to_int64(jobid_list.front.c_str());
```
and further below
```
jcr->impl->previous_jr = JobDbRecord{};
jcr->impl->previous_jr.JobId = str_to_int64(jobid_list.back.c_str());
```